### PR TITLE
[tests] Speed up test_chunkteacher

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
Speed up some slower basic unit tests, mostly by removing a combinatorial explosion of options that isn't really necessary. These tests are also somewhat flaky but hopefully this also helps with that.

**Testing steps**
Before:
```
29.17s call     tests/test_chunkteacher.py::TestNumExamples::test_slow_batchsort
26.23s call     tests/test_chunkteacher.py::TestNumExamples::test_slow_dynb
25.44s call     tests/test_chunkteacher.py::TestNumExamples::test_mp_slow_batchsort
22.49s call     tests/test_chunkteacher.py::TestBackgroundPreprocessorNumExamples::test_mp_slow_dynb
22.28s call     tests/test_chunkteacher.py::TestBackgroundPreprocessorNumExamples::test_mp_slow_batchsort
18.69s call     tests/test_chunkteacher.py::TestBackgroundPreprocessorNumExamples::test_slow_dynb
18.63s call     tests/test_chunkteacher.py::TestBackgroundPreprocessorNumExamples::test_slow_batchsort
12.64s call     tests/test_chunkteacher.py::TestNumExamples::test_slow_bs3
12.55s call     tests/test_chunkteacher.py::TestNumExamples::test_slow_bs2
12.53s call     tests/test_chunkteacher.py::TestNumExamples::test_slow_bs1
======= 60 passed, 113 warnings in 477.21s (0:07:57) =======

```

After:
```
$ pytest tests/test_chunkteacher.py
15.95s call     tests/test_chunkteacher.py::TestSlowChunk::test_slow_dynb
10.15s call     tests/test_chunkteacher.py::TestSmallBuffer::test_mp_small_buffer_batchsort
9.12s call     tests/test_chunkteacher.py::TestNumExamples::test_mp_normal_dynb
8.19s call     tests/test_chunkteacher.py::TestNumExamples::test_normal_dynb
8.04s call     tests/test_chunkteacher.py::TestBackgroundPreprocessorNumExamples::test_mp_normal_dynb
7.80s call     tests/test_chunkteacher.py::TestSmallBuffer::test_mp_small_buffer_dynb
5.24s call     tests/test_chunkteacher.py::TestNumExamples::test_mp_normal_bs3
5.10s call     tests/test_chunkteacher.py::TestNumExamples::test_normal_batchsort
4.96s call     tests/test_chunkteacher.py::TestBackgroundPreprocessorNumExamples::test_mp_normal_bs3
4.77s call     tests/test_chunkteacher.py::TestSmallBuffer::test_mp_small_buffer_bs3
======= 22 passed, 113 warnings in 104.89s (0:01:44) =======
```

### CODE REVIEW

File: tests/test_chunkteacher.py [hunk 1]
```diff
@@ -66,14 +65,6 @@ def test_normal_batchsort(self):
             task='integration_tests:chunky', batchsize=2, dynamic_batching='batchsort'
         )
 
-    @testing_utils.skipUnlessGPU
-    def test_mp_normal_bs1(self):
```
Comment (jxmsML) [inline tests/test_chunkteacher.py:70]: why is this removed?

Comment (jxmsML) [reply tests/test_chunkteacher.py:70]: do you think the remaining test would catch sth like `distributed_world_size > 1 ` while `batchsize = 1` (is this sth that is necessary to test?)

Comment (stephenroller) [reply tests/test_chunkteacher.py:70]: No, I don't think they will.  I can bring that one back if you'd like.

**Review by jxmsML — APPROVED** [review]
LGTM!! Thanks for the fix! 🚀
